src/Type/Builder/ObjectTypeConfig.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace GraphQL\Type\Builder;
+
+class ObjectTypeConfig extends Config
+{
+    public function name($name)
+    {
+        $this->addConfig('name', $name, false);
+        return $this;
+    }
+
+    public function description($description)
+    {
+        $this->addConfig('description', $description, false);
+        return $this;
+    }
+
+    public function addField($name, $type, callable $resolve = null, $description = null, ArgsConfig $args = null, callable $complexity = null, $deprecationReason = null)
+    {
```

Comment:
ok no prob for the order of the addField Arguments, also +1 for the `addDeprecatedField.  we must be careful of reusing  fields definition in closure because same instance will be share in interface and object. I'll work on all this latter. Thanks for the review :)